contracts/upgradeable_contracts/modules/factory/TokenProxy.sol

```diff
@@ -77,6 +79,21 @@ contract TokenProxy is Proxy {
         }
     }
 
+    /**
+     * @dev Updates the implementation contract address.
+     * Only the bridge and bridge owner can call this method.
+     * @param _implementation address of the new implementation.
+     */
+    function setImplementation(address _implementation) external {
+        require(msg.sender == bridgeContractAddr || msg.sender == IOwnable(bridgeContractAddr).owner());
+        require(Address.isContract(_implementation));
+        assembly {
+            // EIP 1967
+            // bytes32(uint256(keccak256('eip1967.proxy.implementation')) - 1)
+            sstore(0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc, _implementation)
```

Comment:
does it make sense to emit an event to track possible updates of the implementation? How much will the gas consumption of the token proxy contract deployment operation be increased after introduction of `setImplementation`?